// @honeybadger-io/react/dist/honeybadger-react.js
var HoneybadgerReact = (function (exports, js, React, PropTypes) {
    'use strict';

    function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }

    var js__default = /*#__PURE__*/_interopDefaultLegacy(js);
    var React__default = /*#__PURE__*/_interopDefaultLegacy(React);
    var PropTypes__default = /*#__PURE__*/_interopDefaultLegacy(PropTypes);

    // eslint-disable-next-line @typescript-eslint/ban-types
    class DefaultErrorComponent extends React.Component {
        render() {
            return (React__default["default"].createElement("div", { className: 'error' },
                React__default["default"].createElement("div", null, "An Error Occurred"),
                React__default["default"].createElement("div", null, JSON.stringify(this.props.error, null, 2)),
                React__default["default"].createElement("div", null, this.props.info ? JSON.stringify(this.props.info, null, 2) : '')));
        }
    }
    DefaultErrorComponent.propTypes = {
        error: PropTypes__default["default"].object,
        info: PropTypes__default["default"].object
    };

    class HoneybadgerErrorBoundary extends React.Component {
        constructor(props) {
            super(props);
            this.state = {
                error: null,
                info: null,
                errorOccurred: false
            };
            this.props.honeybadger.afterNotify((error, _notice) => {
                if (!error && this.props.showUserFeedbackFormOnError) {
                    this.props.honeybadger.showUserFeedbackForm();
                }
            });
        }
        static getDerivedStateFromError(error) {
            return { error: error, errorOccurred: true, info: null };
        }
        componentDidCatch(error, errorInfo) {
            this.setState({ errorOccurred: true, error: error, info: errorInfo });
            this.props.honeybadger.notify(error, { context: errorInfo });
        }
        getErrorComponent() {
            return this.props.ErrorComponent
                ? React__default["default"].createElement(this.props.ErrorComponent, this.state)
                : React__default["default"].createElement(DefaultErrorComponent, { ...this.state });
        }
        render() {
            return (React__default["default"].createElement(React__default["default"].Fragment, null, this.state.errorOccurred
                ? this.getErrorComponent()
                : this.props.children));
        }
    }
    HoneybadgerErrorBoundary.propTypes = {
        honeybadger: PropTypes__default["default"].object.isRequired,
        showUserFeedbackFormOnError: PropTypes__default["default"].bool,
        children: PropTypes__default["default"].element,
        ErrorComponent: PropTypes__default["default"].oneOfType([PropTypes__default["default"].element, PropTypes__default["default"].func])
    };
    HoneybadgerErrorBoundary.defaultProps = {
        showUserFeedbackFormOnError: false
    };

    Object.defineProperty(exports, 'Honeybadger', {
        enumerable: true,
        get: function () { return js__default["default"]; }
    });
    exports.HoneybadgerErrorBoundary = HoneybadgerErrorBoundary;

    Object.defineProperty(exports, '__esModule', { value: true });

    return exports;

})({}, Honeybadger, React, PropTypes);
